File: iRHSCBook/RHSCGuestDetailsViewController.m
//
//  RHSCGuestDetailsViewController.m
//  iRHSCBook
//
//  Created by Bruce Hunter on 2014-07-16.
//  Copyright (c) 2014 Richmond Hill Squash Club. All rights reserved.
//

#import "RHSCGuestDetailsViewController.h"

@interface RHSCGuestDetailsViewController ()

@end

@implementation RHSCGuestDetailsViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.guestNameField.text = self.guest.name;
    self.guestEmailField.text = self.guest.email;
    self.guestPhoneField.text = self.guest.phone;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@synthesize delegate;

-(void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
    self.guest.name = self.guestNameField.text;
    self.guest.email = self.guestEmailField.text;
    self.guest.phone = self.guestPhoneField.text;
    [delegate setGuest:self.guest number:self.guestNumber];
}

-(IBAction)clear
{
    self.guestNameField.text = @"";
    self.guestEmailField.text = @"";
    self.guestPhoneField.text = @"";
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: iRHSCBook/RHSCGuestDetailsViewController.swift
//
//  RHSCGuestDetailsViewController.swift
//  iRHSCBook
//
//  Created by Bruce Hunter on 2015-10-30.
//  Copyright © 2015 Richmond Hill Squash Club. All rights reserved.
//

import Foundation
import UIKit

protocol setGuestProtocol {

    func setGuest(guest:RHSCGuest?,  number:UInt16)

}

class RHSCGuestDetailsViewController : UIViewController {

    var guestNumber : UInt16 = 0
    
    var guest : RHSCGuest? = nil
    
    @IBOutlet weak var guestNameField : UITextField? = nil
    @IBOutlet weak var guestEmailField : UITextField? = nil
    @IBOutlet weak var guestPhoneField : UITextField? = nil
    
    var delegate : AnyObject? = nil
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.guestNameField!.text = self.guest!.name
        self.guestEmailField!.text = self.guest!.email
        self.guestPhoneField!.text = self.guest!.phone
    }

    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
    }
    
    @IBAction func done() {
        if self.guestNameField?.text?.characters.count > 5 {
            self.guest!.name = self.guestNameField!.text!
            self.guest!.email = self.guestEmailField!.text!
            self.guest!.phone = self.guestPhoneField!.text!
            if delegate is RHSCBookCourtViewController {
                let deltarget = (delegate as! RHSCBookCourtViewController)
                deltarget.setGuest(self.guest, number: self.guestNumber)
            }
            self.navigationController?.popViewControllerAnimated(true)
        }
    }
    
    @IBAction func clear() {
        self.guestNameField!.text = ""
        self.guestEmailField!.text = ""
        self.guestPhoneField!.text = ""
    }

}
